src/components/FeaturedEventCard.tsx

```diff
@@ -5,34 +5,79 @@ import Location from '@assets/new/icons/Location';
 import People from '@assets/new/icons/People';
 import Time from '@assets/new/icons/Time';
 import Whatsapp from '@assets/new/icons/Whatsapp';
-import { TrainingSupbaseDataType } from '@data/types/trainingTypes/TypeOrthoexTrainingData';
-import { calculateDateDifference, formatDate } from '@utils/index';
+import { TrainingSupbaseDataType, TrainingPromoDataType  } from '@data/types/trainingTypes/TypeOrthoexTrainingData';
+import { calculateDateDifference, formatDate, formatTime } from '@utils/index';
 import Image from 'next/image';
 import React, { useState } from 'react';
 import styled from 'styled-components';
-import { CTALink } from './CTA';
+import CTA, { CTALink } from './CTA';
 import FeaturedEventDialog from './FeaturedEventDialog';
 import { priceFormatter } from './ProductCard';
+import SocialMediaButtons from './shared/SocialMediaButtons';
 
-interface FeaturedEventProp {
+
+type FeaturedEventProp = {
 	userEmail: string;
 	training: TrainingSupbaseDataType;
+	promoData: TrainingPromoDataType[];
 }
 
 enum EventFormat {
 	ONLINE = 'ONLINE',
 	ONSITE = 'ONSITE',
 }
 
-const FeaturedEventCard: React.FC<FeaturedEventProp> = ({ training }) => {
+const FeaturedEventCard: React.FC<FeaturedEventProp> = ({ training, promoData }) => {
+	const websiteUrl = `https://orthoex.ng/trainings#training-${training.id}`
+	const createUrlText = () => {
+		return encodeURI(`Check out this training: ${training.title} at`)
+	}
+	
 	const [panelOpen, setpanelOpen] = useState(false);
 	const [isBookingDialogOpen, setIsBookingDialogOpen] = useState(false);
+	const [promoCode, setPromoCode] = useState('');
+  const [promoNotification, setPromoNotification] = useState('');
+  const [discountedPrice, setDiscountedPrice] = useState<number | null>(null);
+  const [promoIsValid, setPromoIsValid] = useState<boolean>(false);
 
 	const openBookingDialog = () => setIsBookingDialogOpen(true);
 	const closeBookingDialog = () => setIsBookingDialogOpen(false);
 
+	const trainingPrice = discountedPrice ? discountedPrice : training.price;
+
+	const redeemPromoCode = () => {
+    // Check if the entered promo code exists in the promo list
+    const promo = promoData.find((promo) => promo.promo_code === promoCode);
```

Comment:
This promo check should be done in an API route.

Reply:
I've updated it to use api now